subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/hash_checksum_crc32.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/crc.h>
+#include <fs/fs.h>
+#include <mgmt/mgmt.h>
+#include <fs_mgmt/fs_mgmt_config.h>
+#include <fs_mgmt/fs_mgmt_impl.h>
+#include "fs_mgmt/hash_checksum_mgmt.h"
+#include "fs_mgmt/hash_checksum_crc32.h"
+
+#define CRC32_SIZE 4
+
+static int fs_hash_checksum_mgmt_crc32(struct mgmt_ctxt *ctxt, struct fs_file_t *file,
+				       uint8_t *output, size_t *out_len, size_t len)
+{
+	/* Calculate IEEE CRC32 checksum of target file */
+	uint8_t buffer[CONFIG_FS_MGMT_CHECKSUM_CHUNK_SIZE];
+	ssize_t bytes_read = 0;
+	size_t read_size = CONFIG_FS_MGMT_CHECKSUM_CHUNK_SIZE;
+	uint32_t crc32 = 0;
+
+	/* Clear length prior to calculation */
+	*out_len = 0;
+
+	do {
+		if ((read_size + *out_len) >= len) {
+			/* Limit read size to size of requested data */
+			read_size = len - *out_len;
+		}
+
+		bytes_read = fs_read(file, buffer, read_size);
+		crc32 = crc32_ieee_update(crc32, buffer, bytes_read);
+		*out_len += bytes_read;
+
+		if (*out_len >= len) {
```

Comment:
This can be incorporated into `while` loop condition.

Reply:
Done